src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let output = '';
+
+    for (const item of this) {
+      if (item !== null && item !== undefined) {
+        if (this.indexOf(item) !== this.length - 1) {
```

Comment:
If you already started Algorithms and Data structures, you know what algorithm complexity is.
Using `indexOf` inside a loop increases this complexity.

Much easier way is to replace `for ... of ...` loop with indexed loop, and use index to access the array element.

Reply:
I haven't started the course about algorithms yet. Anyway, I'll mind this fact when I'm doing my upcoming tasks. Thank you.

I rewrote the code using for with indexes.